src/codec/mod.rs

```diff
@@ -146,6 +147,14 @@ impl From<Response> for Bytes {
                 data.put_u16(address);
                 data.put_u16(quantity);
             }
+            ReportServerId(server_id, run_indication, additional_data) => {
+                data.put_u8(2 + u8_len(additional_data.len()));
```

Comment:
- What is the maximum length of `additional_data` according to the spec?
- What happens if you pass more data than allowed? The library should reject such a request instead of panicking.

Reply:
- Is the server allowed to respond with empty `additional` data?

Ideally, there would be tests for all those use cases.

Reply:
I just noticed that there don't seem to be range checks for the other requests :sweat_smile: So I guess we could leave it as is for now.